[Feature] spain adds ceuta

gtrabanco opened the pull request:
## Description
* Added Ceuta autonomous city as CE.
* Added local festivities of Melilla (not sure if ML is Melilla, the 2 letter abbreviation is confusing for us; Spanish people).
* Reordered some ifs and arrays to respect alphabetical order.
* Fixed typo with `País Vasco`.

* Not sure why in some strings the province is separated between "" so I did it with all because is due to any future translation.

## Tests
I am not sure why test are not passing locally on my machine

## Other
Melilla local festivities normally are two more important muslims festivities so I add them as it is in Morocco library but.

I made in other branch this changes and some 2022 exceptions but not sure if 2022 changes must be in other branch. So I will add them after this will be approval because I also have to know a little bit about python to know whats happening with tests.

I am not a Python Dev so give me what I have look for when searching in the net info about any error/fail/improvement. Thank you.

dr-prodigy commented:
Hi,
thank you for your work and support.
As you may have noticed locally, it unfortunately isn't currently passing the provided Spain unit tests.
Here is the error
```
                for fest_day, fest_prov in province_days.items():
                    self.assertEqual(
                        date(year, *fest_day) in prov_holidays,
>                       prov in fest_prov,
                    )
E                   AssertionError: True != False
```
(test_spain.py: 141)

I kindly ask you to please review your code and tests to reflect your changes so we can merge it (also please provide test coverage for all your new code).

Thanks in advance, cheers!

gtrabanco commented:
I do not know why I had the error with CB (Cantabria) by the way I added the Cantabria holidays in the test and also I fixed in the `spain.py` because currently the holidays are two 28 Jul & 15 Sep.

Still does not pass the tests... Not sure why is failing 2014-07-28 with CB. I need help to debug this test.

Added the holiday test for Ceuta.

I also added to the tests for the dynamic year holiday in ML (Melilla) which they are muslims holidays Eid Al-Fitr and Eid-Al-Adha (only the first day of these festivities).

gtrabanco commented:
Test was fixed. Should I add 2022 updates in the same PR or should I add another to add 2022 exceptions if any?

In the doc I have added the new prov and also make clear with parenthesis with prov it is each 2 letter format because now it is not clear.
